faiss/utils/distances.cpp

```diff
@@ -102,7 +102,29 @@ void fvec_renorm_L2(size_t d, size_t nx, float* __restrict x) {
 
 namespace {
 
+// Helpers are used in search functions to help specialize various
+// performance-related use cases, such as adding some extra
+// support for a particular kind of IDSelector classes. This
+// may be useful if the lion's share of samples are filtered out.
+
+struct EmptySelectorHelper {
```

Comment:
please call it IDSelectorAll 
Indeed when you delete elements it makes the index empty but IDSelector is also used for searching a subset of ids.

Reply:
I agree. Let me think how to get rid of this redundancy while keeping the code clear